tests/timers.cpp

```diff
@@ -0,0 +1,78 @@
+#include <catch2/catch_all.hpp>
+#include <zmq.hpp>
+
+#include <type_traits>
+#include <thread>
+#include <chrono>
+
+#ifdef ZMQ_CPP11
```

Comment:
You should guard all all this file with `#if defined(ZMQ_CPP11) && defined(ZMQ_HAVE_TIMERS)`